rasa/core/policies/intent_ted_policy.py

```diff
@@ -536,6 +566,115 @@ def _check_unlikely_intent(
 
         return False
 
+    @staticmethod
+    def _collect_label_id_grouped_scores(
+        output_scores: Dict[Text, Union[np.ndarray, Dict[Text, Any]]],
+        label_ids: np.ndarray,
+    ) -> Dict[int, Tuple[List[float], List[float]]]:
+        """Collects similarities predicted for each label id.
+
+        For each `label_id`, we collect similarity scores across
+        all trackers and categorize them into two buckets:
+            1. Similarity scores when `label_id` is the correct label.
+            2. Similarity scores when `label_id` is the wrong label.
+
+        Args:
+            output_scores: Model's predictions for each data point.
+            label_ids: Numerical IDs of labels for each data point.
+
+        Returns:
+            Both buckets of similarity scores grouped by each unique label id.
+        """
+        label_id_scores: Dict[int, List[List[float]]] = defaultdict(list)
+        unique_label_ids = np.unique(label_ids).tolist()
+        if LABEL_PAD_ID in unique_label_ids:
+            unique_label_ids.remove(LABEL_PAD_ID)
+
+        for label_id in unique_label_ids:
+            label_id_scores[label_id] = [[], []]
```

Comment:
I think it would be better to make this a dict with keys like "positive" and "negative" to avoid having to remember which index is which.

Reply:
Refactored 👍

Reply:
Thanks! Can you not also make this line: 

```python
label_id_scores[label_id] = [[], []]
```

into

```python
label_id_scores[label_id] = {POSITIVE_SCORES_KEY: [], NEGATIVE_SCORES_KEY: []}
```

and the corresponding changes to the lines where you append the scores?

Reply:
My bad, I should have done this in the first place.